cvat-ui/src/components/labels-editor/labels-editor.tsx

```diff
@@ -80,26 +95,17 @@ export default class LabelsEditor
 
         if (!prevProps || prevProps.labels !== labels) {
             const transformedLabels = labels.map(transformLabel);
+            const [savedLabels, unsavedLabels] = getFilteredLabels(transformedLabels)
+
             this.setState({
-                savedLabels: transformedLabels
-                    .filter((label: Label) => label.id >= 0),
-                unsavedLabels: transformedLabels
-                    .filter((label: Label) => label.id < 0),
+                unsavedLabels,
+                savedLabels,
             });
         }
     }
 
     private handleRawSubmit = (labels: Label[]): void => {
-        const unsavedLabels = [];
-        const savedLabels = [];
-
-        for (const label of labels) {
-            if (label.id >= 0) {
-                savedLabels.push(label);
-            } else {
-                unsavedLabels.push(label);
-            }
-        }
+        const [savedLabels, unsavedLabels] = getFilteredLabels(labels)
```

Comment:
Why not? 

```js
const savedLabels = labels.filter((label) => label.id >= 0)
const unsavedLabels = labels.filter((label) => label.id < 0)
```

Reply:
Cause you loop through **labels** two times instead of one. This is just about performance

Reply:
@mmarkelov 

I believe in this place performance doesn't matter just because it isn't a bottleneck.